### PR TITLE
Closes #2433 Remove Query Strings: admin misc changes

### LINKED ISSUE #2433: Remove Query Strings Option Removal: Misc Admin Part (2/8)
Sub-card for the epic #2406

This is for misc admin files.

### inc/admin/ui/notices.php
- [x] Deprecate the `rocket_warning_busting_cache_dir_permissions()` and remove the related `add_action`
### inc/classes/subscriber/admin/Deactivation/class-deactivation-intent-subscriber.php
- [ ] Write tests for `activate_safe_mode`
- [x] `activate_safe_mode` method: remove the `remove_query_strings` entry

### PR DESCRIPTION
**Pull request by crystinutzaa**
Sub-card for the epic #2406

This is for misc admin files.

### inc/admin/ui/notices.php
- [x] Deprecate the `rocket_warning_busting_cache_dir_permissions()` and remove the related `add_action`
### inc/classes/subscriber/admin/Deactivation/class-deactivation-intent-subscriber.php
- [x] Write tests for `activate_safe_mode`
- [x] Rename and move class to `inc/Engine/Admin/Deactivation/DeactivationIntent.php`
- [x] `activate_safe_mode` method: remove the `remove_query_strings` entry

### Live tests
- [ ] QA tests

### CODE REVIEW

**Comment by GeekPress** [comment]
@Tabrisrp @crystinutzaa By deleting the notice about busting folder, what about our Google Tracking and Facebook Pixels add-on which are using this folder too? Do our add-on have customer notices?

**Comment by crystinutzaa** [comment]
@GeekPress good question. However it seems that the notice was available only when Remove Query Strings was enabled: `&& ( get_rocket_option( 'remove_query_strings', false ) )`

@Tabrisrp should we move this notice in Busting Factory? Is there already a task for it?

**Comment by Tabrisrp** [comment]
The notice was indeed only displayed if the remove query strings option was enabled. We have the choice to:
- Preserve the current behaviour, i.e. no warning when the FB/Google tracking add-ons are active
- Add notices to the Facebook/Google tracking subscribers as an enhancement of the current behaviour

If we decide to go with the second option, this can be done on the sub-task corresponding to those add-ons. It will need to be added as an additional task there.

**Comment by GeekPress** [comment]
@Tabrisrp Thanks for the analyze.

Let's go with the second solution.

Why?
The « Remove Query Option » is activated by 70% of our users. 

When the « Google Tracking Add-on » is activated, 80% of these users also activated the « Remove Query String Option ».

In most of cases, we can expect these add-on working thanks to the « Remove Query String Option » and the warning we can have when this option is turned on in case of permission issues.

**Comment by Tabrisrp** [comment]
👌 I added 2 new items on the to-do list for the FB & Google tracking sub-task to take care of this.

@crystinutzaa You can go ahead for this sub-task and keep as planned, other changes will be handled in another PR.

**Comment by Tabrisrp** [comment]
@piotrbak This is ready for QA. Things to test:
- You should no longer have a notice about write permissions for the busting folder when it's not writable
- Activating safe mode should ignore the remove query strings option value now

File: inc/classes/ServiceProvider/class-admin-subscribers.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function register() {
 			->withArgument( $this->getContainer()->get( 'settings_page' ) );
 		$this->getContainer()->add( 'deactivation_intent_render', 'WP_Rocket\Admin\Deactivation\Render' )
 			->withArgument( $this->getContainer()->get( 'template_path' ) . '/deactivation-intent' );
-		$this->getContainer()->share( 'deactivation_intent_subscriber', 'WP_Rocket\Subscriber\Admin\Deactivation\Deactivation_Intent_Subscriber' )
+		$this->getContainer()->share( 'deactivation_intent_subscriber', 'WP_Rocket\Engine\Admin\Deactivation\DeactivationIntent' )
```
Comment (hellofromtonya) [inline inc/classes/ServiceProvider/class-admin-subscribers.php]: As we are touching this file too, let's move and rename it:

- file: `inc/Engine/Admin/ServiceProvider.php`
- class name: `ServiceProvider`.

File: tests/Unit/inc/Engine/Admin/Deactivation/DeactivationIntent/activateSafeMode.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	private $deactivation;
+	private $options;
+	private $options_api;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->options_api  = $this->createMock( 'WP_Rocket\Admin\Options' );
+		$this->options      = $this->createMock( 'WP_Rocket\Admin\Options_Data' );
+		$this->deactivation = new DeactivationIntent( $this->createMock( 'WP_Rocket\Admin\Deactivation\Render' ), $this->options_api, $this->options );
```
Comment (hellofromtonya) [inline tests/Unit/inc/Engine/Admin/Deactivation/DeactivationIntent/activateSafeMode.php]: Replace `$this->createMock` with `Mockery::mock` and then update mock assertions.

File: tests/Unit/inc/Engine/Admin/Deactivation/DeactivationIntent/activateSafeMode.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+		$this->options->expects( $this->once() )
+			->method( 'set_values' )
+			->with( $options );
+		$this->options->expects( $this->once() )
+			->method( 'get_options' )
+			->willReturn( $options );
+		$this->options_api->expects( $this->once() )
+			->method( 'set' )
+			->with( 'settings', $options );
```
Comment (hellofromtonya) [inline tests/Unit/inc/Engine/Admin/Deactivation/DeactivationIntent/activateSafeMode.php]:
```suggestion
		$this->options->shouldReceive( 'set_values' )
			->once()
			->with( $options );
		$this->options->shouldReceive( 'get_options' )
			->once()
			->andReturn( $options );
		$this->options_api->shouldReceive( 'set' )
			->once()
			->with( 'settings', $options );
```

**Pull request merged**